ibis/backends/duckdb/tests/test_compiler.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+
+import ibis
+from ibis.backends.duckdb.compiler import DuckDBSQLCompiler
+
+compiler = DuckDBSQLCompiler()
+
+QUERY = """SELECT array_length(CAST(list_value{} AS ARRAY)) AS n"""
+
+
+@pytest.mark.parametrize(
+    "array",
+    [
+        range(10),
+        range(5),
+        [-1, 0, 1],
+    ],
+)
+def test_compiler_array_length(array):
```

Comment:
I thought we had a backend test for array_length, but maybe not. Can we add one instead of doing this?

Reply:
we do, but it was passing for `duckdb` despite the wacky translation.

Reply:
I'm happy to remove this wacky SQL string comparison thing, but I don't think the existing `array_length` tests are going to prevent a regression (this may not be a huge concern)